src/compiler/compiler.cc

```diff
@@ -1063,120 +1065,148 @@ static const uint8* wrap_direct_script_expression(const char* direct_script, Dia
-  }
-  for (int i = segment_start; i < segments.length(); i++) {
-    path_builder.join(segments[i]->data().c_str());
-    if (i < segments.length() - 1) {
-      // Check if that path fails.
-      std::string built = path_builder.buffer();
-      if (!fs->exists(built.c_str()) || !fs->is_directory(built.c_str())) {
-        auto note_node = segments[i];
-        const char* note_message = fs->exists(built.c_str())
-            ? "Not a folder: '%s'"
-            : "Folder does not exist: '%s'";
-        auto note_path = build_error_path(built);
-        note_fun = [=]() {
-          diagnostics()->report_note(note_node,
-                                     note_message,
-                                     note_path.c_str());
-        };
-        have_note = true;
+    auto note_path = build_error_path(failed_path);
+    switch (error_result) {
+      case AddSegmentResult::NOT_A_REGULAR_FILE:
+        diagnostics->report_note(note_node, "Not a regular file: '%s'", note_path.c_str());
         break;
-      }
-    }
-  }
-  int length_after_segments = path_builder.length();
-  if (!have_note) {
-    auto note_node = segments.last();
-    // We need to append '.toit', or duplicate the last segment.
-    path_builder.add(".toit");
-    auto built = path_builder.buffer();
-    // We would have reported an error earlier if the path existed, but wasn't valid.
-    ASSERT(!fs->exists(built.c_str()));
-    auto toit_path = path_builder.buffer();
-    path_builder.reset_to(length_after_segments);
-    auto potential_dir = path_builder.buffer();
-    path_builder.join(segments.last()->data().c_str());
-    path_builder.add(".toit");
-    auto toit_in_dir = path_builder.buffer();
-
-    if (fs->exists(potential_dir.c_str()) && fs->is_directory(potential_dir.c_str())) {
-      auto note_path = build_error_path(potential_dir);
-      // This file must not exist, as we would have reported an error
-      // earlier, otherwise.
-      ASSERT(!fs->exists(toit_in_dir.c_str()) || !fs->is_regular_file(toit_in_dir.c_str()));
-      auto toit_file = std::string(segments.last()->data().c_str()) + ".toit";
-      note_fun = [=]() {
-        diagnostics()->report_note(note_node,
-                                    "Folder '%s' exists, but is missing a '%s' file",
-                                    note_path.c_str(),
-                                    toit_file.c_str());
-      };
-    } else {
-      auto note_path1 = build_error_path(toit_path);
-      auto note_path2 = build_error_path(toit_in_dir);
-      note_fun = [=]() {
-        diagnostics()->report_note(note_node,
-                                    "Missing library file. Tried '%s' and '%s'",
-                                    note_path1.c_str(),
-                                    note_path2.c_str());
-      };
+      case AddSegmentResult::NOT_A_DIRECTORY:
+        diagnostics->report_note(note_node, "Not a folder: '%s'", note_path.c_str());
```

Comment:
It is not a big deal, but I think the notes could be more uniform. Sometimes the path is at the end, sometimes it is in the middle. Sometimes it starts with 'Not' or 'Missing', sometimes it resembles a proper sentence.